### ISSUE TITLE
fails with output "trace trap" when using @Observable macro

### ISSUE THREAD

**Issue by bryanmontz** [peripheryapp/periphery#700]
When I run periphery on my project, I get this output:
```
* Inspecting project...
* Building <project name>...
* Indexing...
[1]    18879 trace trap  periphery scan
```

The project is a small, new visionOS project. It has some custom modules that are dependencies. As long as I don't have the main app's target in the targets list, periphery completes successfully. If I add the main app to the targets list, it fails in the above way. Is there something in my target's configuration or build settings causing issues?

**Comment by bryanmontz** [comment]
I narrowed it down and figured out that the new `@Observable` macro (iOS 17, visionOS 1) is causing this problem with periphery.

**Comment by ileitch** [comment]
Should be fixed in 2.18.0, please let me know if you find otherwise.
